apis/topic/v1alpha1/topic_types.go

```diff
@@ -27,8 +27,91 @@ import (
 
 // TopicParameters are the configurable fields of a Topic.
 type TopicParameters struct {
+	// +kubebuilder:validation:Minimum:=1
 	ReplicationFactor int `json:"replicationFactor"`
+	// +kubebuilder:validation:Minimum:=1
 	Partitions int `json:"partitions"`
+	// +optional
+	// +kubebuilder:validation:Enum:=compact;delete
+	CleanupPolicy []metav1.List `json:"cleanupPolicy"`
+	// +optional
+	// +kubebuilder:validation:Enum:=uncompressed;zstd;lz4;snappy;gzip;producer
+	CompressionType *string `json:"compressionType"`
```

Comment:
Could we just take these configurations as key/value pairs with a type like `map[string]string`, as mentioned [here](https://github.com/twmb/franz-go/issues/103#issuecomment-964879034)? 

AFAIU, this is the approach other tools follow (e.g. [Terraform provider](https://github.com/Mongey/terraform-provider-kafka#example-1)). This would allow us to better (and more easily) reflect what is on Kafka side.